src/main/java/org/jacksonlaboratory/repository/TermRepository.java

```diff
@@ -17,7 +18,8 @@ public interface TermRepository {
 
 	Optional<List<OntologyTerm>> findByTermIdIn(@NotEmpty List<TermId> ids);
```

Comment:
Why is this an `Optional<List<T>>` but on line 22 you return just `List<T>`?

I think it may be a bit better to just return `List<T>` everywhere, especially on line 16, where wrapping the list in `Optional` seems IMO unnecessary. I would use `Optional` if there was an extra situation that could occur during the lookup. However, failures are probably better modeled with an exception, and missing results for a query can also be represented by an empty list.

Unless you know about some framework semantics which I do not know about..